diff --git a/odoo/addons/base/models/ir_module.py b/odoo/addons/base/models/ir_module.py
index aac796b4630..ebdd7dcc422 100644
--- a/odoo/addons/base/models/ir_module.py
+++ b/odoo/addons/base/models/ir_module.py
@@ -570,13 +570,6 @@ class Module(models.Model):
         }

     def _button_immediate_function(self, function):
-        if getattr(threading.currentThread(), 'testing', False):
-            raise RuntimeError(
-                "Module operations inside tests are not transactional and thus forbidden.\n"
-                "If you really need to perform module operations to test a specific behavior, it "
-                "is best to write it as a standalone script, and ask the runbot/metastorm team "
-                "for help."
-            )
         try:
             # This is done because the installation/uninstallation/upgrade can modify a currently
             # running cron job and prevent it from finishing, and since the ir_cron table is locked
